libraries/AP_Motors/AP_MotorsMatrix.h

```diff
@@ -88,10 +91,14 @@ class AP_MotorsMatrix : public AP_MotorsMulticopter {
     float               _yaw_factor[AP_MOTORS_MAX_NUM_MOTORS];  // each motors contribution to yaw (normally 1 or -1)
     float               _thrust_rpyt_out[AP_MOTORS_MAX_NUM_MOTORS]; // combined roll, pitch, yaw and throttle outputs to motors in 0~1 range
     uint8_t             _test_order[AP_MOTORS_MAX_NUM_MOTORS];  // order of the motors in the test sequence
-    motor_frame_class   _last_frame_class; // most recently requested frame class (i.e. quad, hexa, octa, etc)
-    motor_frame_type    _last_frame_type; // most recently requested frame type (i.e. plus, x, v, etc)
 
     // motor failure handling
     float               _thrust_rpyt_out_filt[AP_MOTORS_MAX_NUM_MOTORS];    // filtered thrust outputs with 1 second time constant
     uint8_t             _motor_lost_index;  // index number of the lost motor
+
+    motor_frame_class   _active_frame_class; // active frame class (i.e. quad, hexa, octa, etc)
+    motor_frame_type    _active_frame_type;  // active frame type (i.e. plus, x, v, etc)
+
+    const char*         _frame_class_string = ""; // string representation of frame class
+    const char*         _frame_type_string = "";  //  string representation of frame type
```

Comment:
Maybe set these to UNKNOWN straight-up - or something distinctive?

Reply:
"uninitialized" would be accurate, but I think this wouldn't tell you anything useful since this just means the motors object failed to initialize for some reason...
Actually, for MotorsMatrix, these strings will be set to "UNSUPPORTED" appropriately when setup_motors fails and _intialized_ok returns false.